aptos-move/aptos-vm/src/aptos_vm.rs

```
@@ -392,6 +398,44 @@ impl AptosVM {
         Ok(())
     }
 
+    fn execute_module_initialization<S: MoveResolverExt>(
+        &self,
+        session: &mut SessionExt<S>,
+        gas_status: &mut GasStatus,
+        modules: &ModuleBundle,
+        senders: &[AccountAddress],
+    ) -> VMResult<()> {
+        let init_func_name = ident_str!("init_module");
+        for module_blob in modules.iter() {
+            let args: Vec<Vec<u8>> = senders
+                .iter()
+                .map(|s| MoveValue::Signer(*s).simple_serialize().unwrap())
+                .collect();
+            match CompiledModule::deserialize(module_blob.code()) {
+                Ok(module) => {
+                    let init_function =
+                        session.load_function(&module.self_id(), init_func_name, &[]);
+                    // it is ok to not have init_module function
+                    // init_module function should be (1) private and (2) has no return value
+                    if init_function.is_ok() && verify_module_init_function(&module).is_ok() {
```

this should return error instead of default to Ok(())? if is_err?

---

udpated